packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -366,6 +422,41 @@ export class PeerPool extends EventEmitter {
 		return true;
 	}
 
+	public addOutboundPeer(
+		peerId: string,
+		peerInfo: P2PDiscoveredPeerInfo,
+		socket: SCClientSocket,
+	): boolean {
+		const existingPeer = this.getPeer(peerId);
+
+		if (existingPeer) {
+			// Update the peerInfo from the latest inbound socket.
+			existingPeer.updatePeerInfo(peerInfo);
+			if (existingPeer.state.outbound === ConnectionState.DISCONNECTED) {
+				existingPeer.outboundSocket = socket;
+
+				return false;
+			}
+
+			return false;
+		}
+
+		const peerConfig = {
+			connectTimeout: this._peerPoolConfig.connectTimeout,
+			ackTimeout: this._peerPoolConfig.ackTimeout,
+		};
+		const peer = new Peer(peerInfo, peerConfig, { outbound: socket });
+
+		// Throw an error because adding a peer multiple times is a common developer error which is very difficult to itentify and debug.
```

Comment:
where does it throw an error?

Reply:
Oh, it was a mistake, we don;t need to throw an error here as we are already checking above if it's present in peer pool or not.